modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBDecision.java

```diff
@@ -38,8 +51,21 @@ public MBDecision(JSONArray json) {
      * Example: ["!", boolean]: boolean
      * @return
      */
-    public Expression decisionNot(){
-        return null;
+    private Expression decisionNot() {
+        // validate the arg list
+        if (json.size() != 2) {
+            throwUnexpectedArgumentCount(NOT, 1);
+        }
+        // first arg is the "!" operator
+        final String notOp = parse.get(json, 0);
+        if (!NOT.equals(notOp)) {
+            // not a NOT(!) op
+            throwUnexpectedOperationException(NOT, notOp);
```

Comment:
Shouldn't this logic already be done by whatever is calling `decisionNot()`? Is this just an extra verification?

Reply:
It is unnecessary validation as the only way each of the decision methods is called is via the switch statement, which is already determining which method to call based on the first argument. Removing it.